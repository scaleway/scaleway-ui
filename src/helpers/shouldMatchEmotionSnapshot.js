/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
