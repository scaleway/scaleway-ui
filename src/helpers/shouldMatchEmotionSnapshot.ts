/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
