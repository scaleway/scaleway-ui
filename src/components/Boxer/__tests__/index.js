import React from 'react'
import { Boxer } from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

describe('Boxer', () => {
  test('renders correctly with one child', () => {
    shouldMatchEmotionSnapshot(<Boxer>Test</Boxer>)
  })

  test('renders correctly with 3 childs', () => {
    shouldMatchEmotionSnapshot(
      <Boxer>
        <h1>Header 1</h1>
        <h2>Header 2</h2>
        <p>paragraph</p>
      </Boxer>,
    )
  })
})
