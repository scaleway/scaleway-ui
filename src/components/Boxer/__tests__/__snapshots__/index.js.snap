// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boxer renders correctly with 3 childs 1`] = `
<DocumentFragment>
  <div
    class="emotion-0 emotion-1"
  >
    <h1>
      Header 1
    </h1>
  </div>
  <div
    class="emotion-0 emotion-1"
  >
    <h2>
      Header 2
    </h2>
  </div>
  <div
    class="emotion-0 emotion-1"
  >
    <p>
      paragraph
    </p>
  </div>
</DocumentFragment>
`;

exports[`Boxer renders correctly with one child 1`] = `
<DocumentFragment>
  <div
    class="emotion-0 emotion-1"
  >
    <p>
      Test
    </p>
  </div>
</DocumentFragment>
`;
